sdcm/sct_events/base.py

```diff
@@ -42,7 +44,68 @@
 LOGGER = logging.getLogger(__name__)
 
 
-class SctEventTypesRegistry(Dict[str, Type["SctEvent"]]):
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class ContinuousEventRegistryError(BaseException):
+    pass
+
+
+class ContinuousEventsRegistry(metaclass=Singleton):
+    def __init__(self):
+        self._continuous_events: List[ContinuousEvent] = []
+
+    @property
+    def continuous_events(self):
+        return self._continuous_events
+
+    def add_event(self, event: ContinuousEvent):
+        if not issubclass(type(event), ContinuousEvent):
+            raise ContinuousEventRegistryError(f"Event: {event} is not a ContinuousEvent")
+
+        self.continuous_events.append(event)
+
+    def get_event(self) -> Optional[ContinuousEvent]:
+        return self.continuous_events[0] if len(self._continuous_events) > 0 else None
+
+    def get_event_by_id(self, event_id: Union[uuid.UUID, str]) -> Optional[ContinuousEvent]:
+        found_event = [event for event in self.continuous_events if event.event_id == event_id]
+        if not found_event:
+            LOGGER.info("Couldn't find continuous event with id: {event_id} in registry.".format(event_id=event_id))
+
+        return found_event[0] if found_event else None
```

Comment:
Why do you return first found event only?

Reply:
I'm filtering by uuid here, so it should return at most 1 event (if any). Other getters are for broader criteria like filtering by type or period.